### ISSUE TITLE
Material binding lost when referenced

### ISSUE THREAD

**Issue by kreppene** [Autodesk/arnold-usd#150]
**Describe the bug**
objects loses there material binding when the primitive is **referenced** 

```
        def Xform "Root"
        {
            def "reference1" (
                prepend references = @../my_beautiful_tree.usda@</g_geometry>
            )
            {
            }
        }
```

**To Reproduce**
Setup a usd file that points to another usd that has material binding setup
and render it
![Screenshot_2](https://user-images.githubusercontent.com/7374105/72601251-54525600-3915-11ea-95a2-4ddd9ec32567.png)

**Used Software Versions**
 - Arnold 6.0.1.0
 - USD:  0.19.11
 - Compiler: Visual studio 2017 
 - OS: windows 10

**Comment by sirpalee** [comment]
@kreppene, can you please confirm whether you are using the procedural or the render delegate?

**Comment by kreppene** [comment]
The test above is from the procedural

**Comment by sirpalee** [comment]
@kreppene can you share an example ass/usda file to reproduce the issue?

In my test referencing meshes from other usda files works just fine with the procedural. Where is the material in your `my_beautilful_tree.usda` file? Is it a child primitive of g_geometry?

For example, if you have a layout like this:

```
def Material "my_material" {
    ...
}

def Mesh "g_geometry" {
    rel material:binding = </my_material>
    ...
}
```

If you are only referencing `g_geometry` in your scene, the material assignment will be broken because the material won't exist in the stage.

**Comment by kreppene** [comment]
@sirpalee yes your absolutely right,  my_beautilful_tree.usda had a layout like the example above,

but even if i change it so the materials and the geo "lives" under the same root, i still get problems.

i attach an example
![Screenshot_4](https://user-images.githubusercontent.com/7374105/72969415-b7d0fd80-3dc5-11ea-820b-ec5a1ef0aef1.png)

if i compare the path to the material between the to render i can se that it has changed
/root/aiStandardSurface1 vs /reference2/aiStandardSurface1.
but i guess thats how it is supose to work?, so the problem im guessing is  the way the geo is refering to the shader.


[test_sphere.zip](https://github.com/Autodesk/arnold-usd/files/4102127/test_sphere.zip)

**Comment by sirpalee** [comment]
@kreppene I think I know what's going wrong with your workflow. Let me do some tests and get back to you with a solution.

**Comment by sebastienblor** [comment]
is this still an issue, or was it fixed by #183 ?

**Comment by sirpalee** [comment]
@sebastienblor no, that pull request didn't fix this issue. It's still a problem of references getting lost because we have a flat hierarchy. One of the possible fixes would be to have put everything into a scope based on a parameter.

**Comment by kreppene** [comment]
maby we could add a  flag to the command line tool, so we could select where to "parent" the materials and shaders?

something along these lines
`arnold_to_usd "asset.ass" "asset.usd" -p "/g_geo"`

**Comment by sirpalee** [comment]
This is going to be resolved by using #292 . Closing this ticket down.
